### PR TITLE
OSDOCS-11439: update microshift config defaults

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
Version(s):
4.17+

Issue:
[OSDOCS-11439](https://issues.redhat.com/browse/OSDOCS-11439)

Link to docs preview:
[MicroShift default values](https://80598--ocpdocs-pr.netlify.app/microshift/latest/microshift_configuring/microshift-using-config-tools.html)

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/microshift-default-settings.adoc [hunk 1]
```diff
@@ -46,21 +46,27 @@ ingress:
...
 node:
-  hostnameOverride: "" # <18>
-  nodeIP: "" # <19>
+  hostnameOverride: "" # <19>
+  nodeIP: "" # <20>
+  nodeIPv6: "" # <21>
+storage:
+    driver: "" # <22>
+    optionalCsiComponents:
+        - "" # <23>
```
Comment (copejon) [inline modules/microshift-default-settings.adoc:69]:
```suggestion
storage:
  driver: "" # <22>
  optionalCsiComponents:  # <23>
  - ""
```

File: modules/microshift-default-settings.adoc [hunk 2]
```diff
@@ -75,9 +81,13 @@ node:
...
+<17> A block of virtual IP addresses for Kubernetes services. IP address pool for services. A single entry is supported. This field is immutable after installation.
+<18> The port range allowed for Kubernetes services of type `NodePort`. If not specified, the default range of 30000-32767 is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after the cluster is installed.
+<19> The name of the node. The default value is the hostname. If non-empty, this string is used to identify the node instead of the hostname.
+<20> The IP address of the node. The default value is the IP address of the default route.
+<21> Configurable IPv6 network.
+<22> Default value is empty. Valid values are "none" or "lvms". An empty value or null field defaults to LVMS deployment.
+<23> Default value is empty. An empty array defaults to deploying `snapshot-controller` and `snapshot-webhook`.Expected values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. An entry of `none` is mutually exclusive with all other values.
```
Comment (copejon) [inline modules/microshift-default-settings.adoc:93]:
```suggestion
<23> Array. Default value is null or an empty array. A null or empty array defaults to deploying `snapshot-controller` and `snapshot-webhook`. Expected values are `csi-snapshot-controller`, `csi-snapshot-webhook`, or `none`. An entry of `none` is mutually exclusive with all other values.
```

File: modules/microshift-default-settings.adoc (hunk 1 above)
Comment (copejon) [reply modules/microshift-default-settings.adoc:69]: Moved the number since it's the `optionalCsiComponents` field that's actually being described below.

**Review by copejon — CHANGES_REQUESTED** [review]

File: modules/microshift-default-settings.adoc [hunk 3]
```diff
@@ -75,9 +81,13 @@ node:
...
-<14> The locations on the file system to scan for `kustomization` files to use to load manifests. Set to a list of paths to scan only those paths. Set to an empty list to disable loading manifests. The entries in the list can be glob patterns to match multiple subdirectories.
-<15> A block of IP addresses from which pod IP addresses are allocated. This field is immutable after installation.
-<16> A block of virtual IP addresses for Kubernetes services. IP address pool for services. A single entry is supported. This field is immutable after installation.
-<17> The port range allowed for Kubernetes services of type `NodePort`. If not specified, the default range of 30000-32767 is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after the cluster is installed.
-<18> The name of the node. The default value is the hostname. If non-empty, this string is used to identify the node instead of the hostname.
-<19> The IP address of the node. The default value is the IP address of the default route.
+<14> Optional parameter for configuring low latency. Default value is null.
```
Comment (pmtk) [inline modules/microshift-default-settings.adoc:84]:
```suggestion
<14> Optional parameter for configuring kubelet. Default value is null.
```
Technically someone could configure anything related to kubelet using that field. We don't really want that, but saying it's for "configuring low latency" feels off.

File: modules/microshift-default-settings.adoc [hunk 4]
```diff
@@ -75,9 +81,13 @@ node:
...
+<15> The locations on the file system to scan for `kustomization` files to use to load manifests. Set to a list of paths to scan only those paths. Set to an empty list to disable loading manifests. The entries in the list can be glob patterns to match multiple subdirectories.
+<16> A block of IP addresses from which pod IP addresses are allocated. This field is immutable after installation.
+<17> A block of virtual IP addresses for Kubernetes services. IP address pool for services. A single entry is supported. This field is immutable after installation.
```
Comment (pacevedom) [inline modules/microshift-default-settings.adoc:87]: These two now allow adding one more entry for the other IP family. It is true though that the first entry of the list must remain unchanged.

File: modules/microshift-default-settings.adoc [hunk 5]
```diff
@@ -75,9 +81,13 @@ node:
...
+<15> The locations on the file system to scan for `kustomization` files to use to load manifests. Set to a list of paths to scan only those paths. Set to an empty list to disable loading manifests. The entries in the list can be glob patterns to match multiple subdirectories.
+<16> A block of IP addresses from which pod IP addresses are allocated. IPv4 is the default. Dual-stack entries are supported. The first entry in this field is immutable after installation.
+<17> A block of virtual IP addresses for Kubernetes services. IP address pool for services. IPv4 is the default. Dual-stack entries are supported. The first entry in this field is immutable after installation.
+<18> The port range allowed for Kubernetes services of type `NodePort`. If not specified, the default range of 30000-32767 is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after the cluster is installed.
+<19> The name of the node. The default value is the hostname. If non-empty, this string is used to identify the node instead of the hostname.
+<20> The IP address of the node. The default value is the IP address of the default route.
```
Comment (dhensel-rh) [inline modules/microshift-default-settings.adoc:90]: Small nit. Maybe use IPv4 in place of IP here

File: modules/microshift-default-settings.adoc [hunk 6]
```diff
@@ -75,9 +81,13 @@ node:
...
-<14> The locations on the file system to scan for `kustomization` files to use to load manifests. Set to a list of paths to scan only those paths. Set to an empty list to disable loading manifests. The entries in the list can be glob patterns to match multiple subdirectories.
-<15> A block of IP addresses from which pod IP addresses are allocated. This field is immutable after installation.
-<16> A block of virtual IP addresses for Kubernetes services. IP address pool for services. A single entry is supported. This field is immutable after installation.
-<17> The port range allowed for Kubernetes services of type `NodePort`. If not specified, the default range of 30000-32767 is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after the cluster is installed.
-<18> The name of the node. The default value is the hostname. If non-empty, this string is used to identify the node instead of the hostname.
-<19> The IP address of the node. The default value is the IP address of the default route.
+<14> Parameter for configuring kubelet. Used for low latency configuration. Default value is null.
```
Comment (jneczypor) [inline modules/microshift-default-settings.adoc:84]:
```suggestion
<14> Parameter for configuring kubelet. Used for low-latency configuration. Default value is null.
```

**Review by jneczypor — COMMENTED** [review]
Just one small change. Otherwise, LGTM.

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**